evm-contracts/src/v0.8/dev/Anchors/UniswapAnchoredView.sol

```diff
@@ -0,0 +1,388 @@
+ *
+ * This contract differs from the original in several ways:
+ *   - Instead of a reporter posting prices, an AccessControlledOffChainAggregator contract submits prices via the validate function defined in the AggregatorValidatorInterface.
+ *   - Each configured TokenConfig has its own reporter (aggregator) instead of a single reporter. This reporter is responsible for posting prices for a particular token.
+ *   - This contract has an owner (community multisig) with a single privilege: the ability to invalidate a specific reporter.
+ *       Each reporter is invalidated individually, affecting one market at a time rather than every single market. This is because each reporter will consist of a unique decentralised price feed with independent operators.
+ *   - SafeMath functions are removed since this is written for Solidity version ^0.8.0.
+ */
+contract UniswapAnchoredView is AggregatorValidatorInterface, UniswapConfig, ConfirmedOwner {
+  using FixedPoint for *;
+
+  /// @notice The number of wei in 1 ETH
+  uint public constant ethBaseUnit = 1e18;
+
+  /// @notice A common scaling factor to maintain precision
+  uint public constant expScale = 1e18;
+
+  /// @notice The highest ratio of the new price to the anchor price that will still trigger the price to be updated
+  uint public immutable upperBoundAnchorRatio;
+
+  /// @notice The lowest ratio of the new price to the anchor price that will still trigger the price to be updated
+  uint public immutable lowerBoundAnchorRatio;
+
+  /// @notice The minimum amount of time in seconds required for the old uniswap price accumulator to be replaced
+  uint public immutable anchorPeriod;
+
+  /// @notice Official prices by symbol hash
+  mapping(bytes32 => uint) public s_prices;
+
+  /// @notice The old observation for each symbolHash
+  mapping(bytes32 => Observation) public s_oldObservations;
+
+  /// @notice The new observation for each symbolHash
+  mapping(bytes32 => Observation) public s_newObservations;
+
+  /// @notice Reporters that have been invalidated by the owner
+  mapping(address => bool) public s_reporterInvalidated;
+
+  /// @notice The event emitted when new prices are posted but the stored price is not updated due to the anchor
+  event PriceGuarded(
+    bytes32 symbolHash,
+    uint reporter,
+    uint anchor
+  );
+
+  /// @notice The event emitted when the stored price is updated
+  event PriceUpdated(
+    bytes32 symbolHash,
+    uint price
+  );
+
+  /// @notice The event emitted when anchor price is updated
+  event AnchorPriceUpdated(
+    bytes32 symbolHash,
+    uint anchorPrice,
+    uint oldTimestamp,
+    uint newTimestamp
+  );
+
+  /// @notice The event emitted when the uniswap window changes
+  event UniswapWindowUpdated(
+    bytes32 indexed symbolHash,
+    uint oldTimestamp,
+    uint newTimestamp,
+    uint oldPrice,
+    uint newPrice
+  );
+
+  /// @notice The event emitted when reporter invalidates itself
+  event ReporterInvalidated(
```

Comment:
Can this happen in our case?

Reply:
To clarify, the owner would invalidate the reporter, not the reporter itself. Correct?